### PR TITLE
SWARM-771: Provide a zipkin fraction

### PR DESCRIPTION
**Pull request by heiko-braun**
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

---
## Motivation

In order to integrate with Hawkular BTM, zipkin instrumentation is needed
## Modifications

This introduces a zipkin fraction that adds the instrumentation to server side JAX-RS resources
## Result

Zipkin instrumented JAX-RS resources can now send the reporting data to a zipkin backend

### CODE REVIEW

**Comment by heiko-braun** [comment]
This change was initially motivated by https://issues.jboss.org/browse/SWARM-624

File: fractions/zipkin-jaxrs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# ignore .svn metadata files
```
Comment (gastaldi) [inline fractions/zipkin-jaxrs/.gitignore]: Is this file necessary?

File: fractions/zipkin-jaxrs/src/main/java/org/wildfly/swarm/jaxrs/btm/runtime/BraveService.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    public Brave get() {
+        return this.brave;
+    }
+
+    @Override
+    public void start(StartContext startContext) throws StartException {
+        System.out.println("Zipkin BTM services started: " + this.brave);
```
Comment (gastaldi) [inline fractions/zipkin-jaxrs/src/main/java/org/wildfly/swarm/jaxrs/btm/runtime/BraveService.java]: Use Messages instead of System.out?

**Review by gastaldi — APPROVED** [review]
It would be nice to have some tests, otherwise it looks good to me

File: fractions/zipkin-jaxrs/.gitignore (hunk 1 above)
Comment (kenfinnigan) [reply fractions/zipkin-jaxrs/.gitignore]: +1 shouldn't need separate .gitignore

File: fractions/zipkin-jaxrs/pom.xml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    <dependency>
+      <groupId>org.wildfly.swarm</groupId>
+      <artifactId>jaxrs</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.wildfly.swarm</groupId>
+      <artifactId>msc</artifactId>
+    </dependency>
+    <dependency>
```
Comment (kenfinnigan) [inline fractions/zipkin-jaxrs/pom.xml]: undertow shouldn't need to be specified, and jaxrs is present twice

File: fractions/zipkin-jaxrs/src/main/java/org/wildfly/swarm/jaxrs/btm/runtime/WebXMLAdapter.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    private final static String SERVER_SIDE_FILTERS =
+            ServerRequestInterceptor.class.getName()+","
+                    +ServerResponseInterceptor.class.getName();
+
+    @Override
+    public void prepareArchive(Archive<?> archive) {
+        if(archive instanceof JAXRSArchive) {
+            JAXRSArchive jaxrsArchive = archive.as(JAXRSArchive.class);
+            jaxrsArchive.findWebXmlAsset().setContextParam(
```
Comment (kenfinnigan) [inline fractions/zipkin-jaxrs/src/main/java/org/wildfly/swarm/jaxrs/btm/runtime/WebXMLAdapter.java:23]: I always thought @Provider on the class was enough to trigger it?

File: fractions/zipkin-jaxrs/src/test/java/org/wildfly/swarm/jaxrs/btm/ZipkinTest.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package org.wildfly.swarm.jaxrs.btm;
+
+/**
+ * @author Heiko Braun
+ * @since 17/10/16
+ */
+public class ZipkinTest {
```
Comment (gastaldi) [inline fractions/zipkin-jaxrs/src/test/java/org/wildfly/swarm/jaxrs/btm/ZipkinTest.java]: Looks like this class is not useful ?

**Comment by gastaldi** [comment]
Reviewing locally. Will merge after applying some changes

**Comment by heiko-braun** [comment]
Thanks @gastaldi

**Comment by gastaldi** [comment]
Merged in https://github.com/wildfly-swarm/wildfly-swarm/commit/4040187dffc05a2f95ce5614b7510d8a4521e842
